clone/cmd/ctverify/ctverify.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ctclone is a one-shot tool for downloading entries from a CT log.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian-examples/clone/internal/logdb"
+	"github.com/google/trillian-examples/clone/internal/verify"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+var (
+	mysqlURI = flag.String("mysql_uri", "", "URL of a MySQL database to clone the log into. The DB should contain only one log.")
+)
+
+func main() {
+	flag.Parse()
+
+	if len(*mysqlURI) == 0 {
+		glog.Exit("Missing required parameter 'mysql_uri'")
+	}
+	db, err := logdb.NewDatabase(*mysqlURI)
+	if err != nil {
+		glog.Exitf("Failed to connect to database: %q", err)
+	}
+
+	bs, err := ioutil.ReadAll(os.Stdin)
```

Comment:
TODO? would be nice to have this be able to read a file too (and perhaps default to `stdin`)

Reply:
I'm not convinced that the feature being added or not would be changed by the presence of a TODO. I left it this way because it's easy to pipe a file into this if that's where the checkpoint is, but one could also pipe the results of a SQL query or anything else. If files really do become the standard place checkpoints are stored, then I would support a file flag being added.

Reply:
Oke doke